base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -47,9 +47,9 @@ impl LinearWeightedMovingAverage {
             .target_difficulties
             .iter()
             .skip(1)
-            .fold(0u64, |difficulty, (_, d)| difficulty + d.as_u64());
+            .fold(0u128, |difficulty, (_, d)| difficulty + d.as_u64() as u128);
 
-        let ave_difficulty = difficulty as f64 / n as f64;
+        let ave_difficulty = (difficulty / n as u128) as u64;
```

Comment:
Perhaps add a comment stating why you are doing this. 

Also, maybe just set `n` as a `u128` when you declare it

Reply:
You should also add a unit test to prove that this doesn't overflow